static/app/components/replays/canvasReplayerPlugin.tsx

```diff
@@ -102,45 +220,59 @@ export function CanvasReplayerPlugin(events: eventWithTime[]) {
      */
     handler: async (
       e: eventWithTime,
-      _isSync: boolean,
+      isSync: boolean,
       {replayer}: {replayer: Replayer}
     ) => {
-      if (
-        e.type === EventType.IncrementalSnapshot &&
-        e.data.source === IncrementalSource.CanvasMutation
-      ) {
-        const source = replayer.getMirror().getNode(e.data.id);
-        const target =
-          canvases.get(e.data.id) ||
-          (source && cloneCanvas(e.data.id, source as HTMLCanvasElement));
-
-        // No canvas found for id... this isn't reliably reproducible and not
-        // exactly sure why it flakes. Saving as metric to keep an eye on it.
-        if (!target) {
-          Sentry.metrics.increment('replay.canvas_player.no_canvas_id');
-          return;
+      // isSync = true means it is fast forwarding vs playing
+      // nothing to do when fast forwarding since canvas mutations for us are
+      // image snapshots and do not depend on past events
+      if (isSync || !isCanvasMutationEvent(e)) {
+        if (isSync) {
+          // Set this to -1 to indicate that we will need to search
+          // `canvasMutationEvents` for starting point of preloading
+          //
+          // Only do this when isSync is true, meaning there was a seek
+          nextPreloadIndex = -1;
         }
+        prune(e);
+        return;
+      }
 
-        await canvasMutation({
-          event: e,
-          mutation: e.data,
-          target: target as HTMLCanvasElement,
-          imageMap,
-          canvasEventMap,
-          errorHandler: (err: unknown) => {
-            if (err instanceof Error) {
-              Sentry.captureException(err);
-            } else {
-              Sentry.metrics.increment('replay.canvas_player.error_canvas_mutation');
-            }
-          },
-        });
-
-        const img = containers.get(e.data.id);
-        if (img) {
-          img.src = target.toDataURL();
-        }
+      preload(e);
+
+      const source = replayer.getMirror().getNode(e.data.id);
+      const target =
+        canvases.get(e.data.id) ||
+        (source && cloneCanvas(e.data.id, source as HTMLCanvasElement));
+
+      // No canvas found for id... this isn't reliably reproducible and not
+      // exactly sure why it flakes. Saving as metric to keep an eye on it.
+      if (!target) {
```

Comment:
I suspect this could happen if the id isn't found in the mirror. That happens sometimes when seeking really fast and the play doesn't keep up... that message shows up in the console, something like "id not found" over and over and the playback is broken.